zrcs/zrc-2.md

```diff
@@ -0,0 +1,372 @@
+| ZRC | Title                            | Status | Type     | Author                             | Created (yyyy-mm-dd) | Updated (yyyy-mm-dd) |
+| --- | -------------------------------- | ------ | -------- | ---------------------------------- | -------------------- | -------------------- |
+| 1   | Standard for Fungible Tokens | Draft  | Ready | Gareth Mensah <gareth@zilliqa.com> | 2019-11-18           | 2019-11-18           |
+
+## I. What are Fungible Tokens?
+
+The Fungible Token is an open standard for creating currencies. Fungibility is the property of a good or a commodity whose individual units are essentially interchangeable, and each of its parts is indistinguishable from another part.
+
+## II. Abstract
+
+ZRC-2 defines a minimum interface a smart contract must implement to allow fungible tokens to be managed, tracked, owned, and traded.
+
+## III. Motivation
+
+A standard for fungible tokens can serve as an interface for developers to create currencies, utility tokens or stable coins. Generally, fungible tokens can be used to represent interchangeable, identical and divisable assets as tokens.
+
+## IV. Specification
+
+The fungible token contract specification describes:
+
+1. the global error codes to be declared in the library part of the contract;
+2. the names and types of the immutable and mutable variables (aka `fields`);
+3. the transitions that will allow the changing of values of the mutable variables;
+4. the events to be emitted by them.
+
+### A. Roles
+
+| Name              | Description                                                                                                                                                                                                                      |     |
+| ----------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |--|
+| `contractOwner`   | The owner of the contract initialized by the creator of the contract.                                                                                                                                                            |
+| `tokenOwner`      | A user (identified by an address) that owns a token tied to a tokenId.                                                                                                                                                           |
```

Comment:
Fungible tokens doesn't have the concept of `tokenId` right? Looks like this has come in why copying over from non-fungible tokens.